lib/instrumentation/express-utils.js

```diff
@@ -28,12 +29,17 @@ function getStackPath (req) {
 
 // This function is also able to extract the path from a Restify request as
 // it's storing the route name on req.route.path as well
-function getPathFromRequest (req, useBase) {
+function getPathFromRequest (req, useBase, usePathAsTransactionName) {
   // Static serving route
   if (req[symbols.staticFile]) {
     return 'static file'
   }
 
+  // Enable usePathAsTransactionName config
+  if (usePathAsTransactionName) {
```

Comment:
To do exactly what the docs describe, this should probably be moved up above the first if-sentence, so that even static routes will have their name set by the pathname